### ISSUE TITLE
awesome work!

### ISSUE THREAD

**Issue by bewest** [ps2/minimed_rf#1]
Howdy,

just wanted to say hello, and this looks awesome.

Looks like you did most of this on your own?

It would be great to work on shared unit tests, especially for history.
FWIW, you should be able to add support for the carelink stick fairly easily.
I'm pretty sure @bustavo is going to like the ruby decoding history.

* python: https://github.com/bewest/decoding-carelink
* js: https://github.com/bewest/comlink2-uart
* java: https://github.com/nightscout/android-uploader/tree/minimed/core/src/main/java/com/nightscout/core/drivers/Medtronic

And now we have ruby.
There's a group that discusses this stuff in https://gitter.im/nightscout/intend-to-bolus, we'd love for you to visit.
Also, you can compare symbol tables against @jberian's work: https://github.com/jberian/mmcommander, https://gist.github.com/bewest/6d05f80c932cc282ecfd.

Looking forward to discussing more!

~@bewest

**Comment by bewest** [comment]
https://github.com/ps2/minimed_rf/blob/master/lib/minimed_rf/messages/pump_status.rb#L56-L58

Is this for `523`?  What is this `0x04` `PumpStatus` thing? https://github.com/ps2/minimed_rf/blob/master/lib/minimed_rf/messages/message_type_map.rb#L5


Is this the RF message type?  How did you learn more about these?  https://github.com/bewest/decoding-carelink/blob/master/decocare/stick.py#L341  Are these similar to the `PumpCommand` (`0xa7`) vs `MeterNow` type of responses, or specific to the mysentry or something, or more like the `ReadHistoryData` (`0x80`) commands?

Very interesting stuff, I'm wondering if the carelink usb stick can send/receive these other command types, especially the meter communications.

I notice a few unused bits in a few places, it would be cool to compare notes somehow.  I've thought about writing little script to dump table/list of record types to assist preparing documentation, but not had a lot of time to do that yet.

**Comment by StephenBlackWasAlreadyTaken** [comment]
soooooo happy its ruby!

**Comment by ps2** [comment]
The active insulin is on both my test pump, which is a 523, and my daughter's pump, which is a 530G. Was super happy to discover this in the data, as it is not displayed on the mysentry.

The 0x04 message type are the messages that the pump sends to the mysentry.

Yes, the first byte (what I call message type) is where you would find 0xa7 when carelink is downloading from the pump.  The 0x80 comes later in the packet.  Here is a full dump of a convo with the pump using the carelink java app and my usb meter stick, with the 0x80 history section highlighted: https://gist.github.com/ps2/338b4ba5d43bf0199c55#file-gistfile1-txt-L614-L648

**Comment by bewest** [comment]
Have you tested key press commands/temp basal?

So your stick is pretending to be a sentry?  Is that how it works?  And the pump naturally sends stuff to it on regular basis?

**Comment by bustavo** [comment]
Ruby ruby ruby ruby ruby :smile:

**Comment by ps2** [comment]
Yes, the device I built mimics the MySentry, and receives updates from the pump every 5 minutes (which is the timing of the enlite sensor). It then forwards the data via bluetooth to a cellphone (iOS) for uploading into the cloud, which is currently my own own server software, but I should probably switch over to the nightscout platform, or at least write something to bridge it.

**Comment by ps2** [comment]
hmm. The gitter.im site wants access to my organizations for me to chat there. Looks like the organizations have to explicitly deny this.

**Comment by inv4der** [comment]
Hi, awesome work, do you know If I can intercept the minimed enlite data directly?

**Comment by ps2** [comment]
Yes, you can receive enlite data; you might need to tweak the frequency, as it is a bit different from the pump, but the rf encoding is the same.  Packet format is different than the pump, so you’d need to decode that as well.

**Comment by inv4der** [comment]
Do you know which frequency is?
